packages/di/README.md

```diff
@@ -74,6 +74,25 @@ const car2 = factory2.get(Car);
 console.log(car2.accelerate(), car2.tires().size);
 ```
 
+#### Factories
+
+In addition to defining providers with classes you can also use factory functions.
+
+```ts
+abstract class Logger {
+  abstract log(...args: any[]): void
+}
+
+const app = new Injector([
+  {
+    provide: Logger,
+    factory() {
+      return console 
```

Comment:
minor: console is not defined (yet)

Maybe we also want to showcase the decorator usecase here?

Reply:
This example is just using the default js console. Wanted to demonstrate you can return anything that implements your token

Reply:
you are right, ofc console is defined! (in all JS runtimes I've tested so far)

Reply:
And I did not know about abstract classes being valid as ProviderToken! 
That closes the circle back in #961:  an abstract class will not / cannot be instanciated and thereby does not provide a fallback provider.

Reply:
yeah abstract classes in typescript can be treated as tokens because they are compiled to actual js code where an interface does not